db/compaction/compaction_picker.cc

```diff
@@ -1078,11 +1078,27 @@ Status CompactionPicker::SanitizeCompactionInputFiles(
 
   Status s = SanitizeCompactionInputFilesForAllLevels(input_files, cf_meta,
                                                       output_level);
+  if (!s.ok()) {
+    return s;
+  }
 
+  s = GetCompactionInputsFromFileNumbers(converted_input_files, input_files,
+                                         vstorage, CompactionOptions());
   if (!s.ok()) {
     return s;
   }
 
+  assert(converted_input_files->size() > 0);
+  if (FilesRangeOverlapWithCompaction(
```

Comment:
Does anyone know why we allow output_level=0 to short-circuit the assertion check here https://github.com/facebook/rocksdb/blob/933ac0e05ce8c14847d0c67160c3c5869eae9656/db/compaction/compaction_picker.cc#L353????


```
assert(output_level == 0 ||
         !FilesRangeOverlapWithCompaction(
             input_files, output_level,
             Compaction::EvaluatePenultimateLevel(vstorage, ioptions_,
                                                  start_level, output_level)));
```

Reply:
Presumably because L0 files are generally allowed to overlap

Reply:
okay then I should align that here too - fixing